app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/TeamsLocationsMapActivity.java

```diff
@@ -234,9 +235,11 @@ private void setupIntent(Intent intent) {
         Uri data = intent.getData();
         Bundle extras = intent.getExtras();
         if (data != null && !mIsConsumedUriIntent) {
-            String teamNumberParameterValue = data.getQueryParameter(Constants.URL_MAP_TEAM_NUMBER_PARAM);
-            if (teamNumberParameterValue != null) {
-                changeTeamFromIntent(teamNumberParameterValue);
+            List<String> pathSegments = data.getPathSegments();
+            String teamPathValue = pathSegments.get(0);
+            String teamNumberPathValue = pathSegments.get(1);
+            if (teamPathValue.equals(Constants.URL_MAP_TEAM_NUMBER_PATH) && teamNumberPathValue != null) {
```

Comment:
```suggestion
            if (Constants.URL_MAP_TEAM_NUMBER_PATH.equals(teamPathValue) && teamNumberPathValue != null) {
```

Zabezpiecza przed NPE.